src/main/webapp/app/modules/administration/user-management/user-management-update.tsx

```diff
@@ -110,7 +123,16 @@ export const UserManagementUpdate = (props: IUserManagementUpdateProps) => {
                 name={'authorities'}
                 options={authorityOptions}
                 defaultValue={[{ label: USER_AUTHORITY.ROLE_USER, value: USER_AUTHORITY.ROLE_USER }]}
+                onChange={(selectedOption, actionMeta) => {
+                  if (actionMeta.action === 'remove-value' && actionMeta.removedValue?.value === USER_AUTHORITY.ROLE_DEV) {
+                    setIsDev(false);
+                  }
+                  if (actionMeta.action === 'select-option' && selectedOption.some(option => option.value === USER_AUTHORITY.ROLE_DEV)) {
+                    setIsDev(true);
+                  }
+                }}
               />
+              {isDev && <ValidatedSelect isMulti label="Feature Flag" name={'featureFlags'} options={featureFlagOptions} />}
```

Comment:
Instead of isDev state variable, I would just check if the user has ROLE_DEV authority. See my comment on line 154.